main.go

```diff
@@ -1,77 +1,83 @@
 	"net/http"
-	"path/filepath"
+	"os"
+	"os/signal"
+	"syscall"
 	"time"
 
-	"github.com/codeready-toolchain/registration-service/static"
-
-	"github.com/gorilla/mux"
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	"github.com/codeready-toolchain/registration-service/pkg/registrationserver"
 )
 
-// spaHandler implements the http.Handler interface, so we can use it
-// to respond to HTTP requests. The path to the static directory and
-// path to the index file within that static directory are used to
-// serve the SPA in the given static directory.
-type spaHandler struct {
-	Assets http.FileSystem
-}
+func main() {
+	// Parse flags
+	var configFilePath string
+	flag.StringVar(&configFilePath, "config", "", "path to the config file to read (if none is given, defaults will be used)")
+	flag.Parse()
+
+	// Override default -config switch with environment variable only if -config
+	// switch was not explicitly given via the command line.
+	configSwitchIsSet := false
+	flag.Visit(func(f *flag.Flag) {
+		if f.Name == "config" {
+			configSwitchIsSet = true
+		}
+	})
+	if !configSwitchIsSet {
+		if envConfigPath, ok := os.LookupEnv(configuration.EnvPrefix + "_CONFIG_FILE_PATH"); ok {
+			configFilePath = envConfigPath
+		}
+	}
 
-// ServeHTTP inspects the URL path to locate a file within the static dir
-// on the SPA handler. If a file is found, it will be served. If not, the
-// file located at the index path on the SPA handler will be served. This
-// is suitable behavior for serving an SPA (single page application).
-func (h spaHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	// get the absolute path to prevent directory traversal
-	path, err := filepath.Abs(r.URL.Path)
+	srv, err := registrationserver.New(configFilePath)
 	if err != nil {
-		// no absolute path, respond with a 400 bad request and stop
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		return
+		panic(err.Error())
 	}
 
-	// check if the file exists in the assets
-	_, err = h.Assets.Open(path)
+	err = srv.SetupRoutes()
 	if err != nil {
-		// file does not exist, redirect to index
-		log.Printf("File %s does not exist.", path)
-		http.Redirect(w, r, "/index.html", http.StatusSeeOther)
-		return
+		panic(err.Error())
 	}
 
-	// otherwise, use http.FileServer to serve the static dir
-	http.FileServer(h.Assets).ServeHTTP(w, r)
-}
+	routesToPrint, err := srv.GetRegisteredRoutes()
+	if err != nil {
+		panic(err.Error())
+	}
+	srv.Logger().Print(routesToPrint)
 
-// HealthCheckHandler returns a default heath check result.
-func HealthCheckHandler(w http.ResponseWriter, r *http.Request) {
-	// default handler for system health
-	w.Header().Set("Content-Type", "application/json")
-	json.NewEncoder(w).Encode(map[string]bool{"alive": true})
-}
+	// listen concurrently to allow for graceful shutdown
+	go func() {
+		srv.Logger().Printf("Listening on %q...", srv.Config().GetHTTPAddress())
+		if err := srv.HTTPServer().ListenAndServe(); err != nil {
```

Comment:
AFAIK, this has removed the SSL support...see `master`. I guess there should be a config directive each for certificate and key files and this line here should check whether to run in http or https mode. I also added an `insecure` config in `master`. The default would be https mode, with bail out when the cert and key files could not be found. If someone wants to run in http mode, that must be explicitly given. Makes it a bit more secure.